File: components/credentials/src/main/java/org/cloudfoundry/credhub/audit/entities/GetCredential.java
package org.cloudfoundry.credhub.audit.entities;

import java.util.Objects;

import org.apache.commons.lang3.builder.EqualsBuilder;
import org.cloudfoundry.credhub.audit.OperationDeviceAction;
import org.cloudfoundry.credhub.audit.RequestDetails;

public class GetCredential implements RequestDetails {
  private String name;
  private Integer versions;
  private Boolean current;

  public GetCredential(final String credentialName, final Integer numberOfVersions, final boolean current) {
    super();
    name = credentialName;
    versions = numberOfVersions;
    this.current = current;
  }

  public GetCredential() {
    super();

  }

  @Override
  public boolean equals(final Object o) {
    if (this == o) {
      return true;
    }

    if (o == null || getClass() != o.getClass()) {
      return false;
    }

    final GetCredential that = (GetCredential) o;

    return new EqualsBuilder()
      .append(name, that.name)
      .append(versions, that.versions)
      .append(current, that.current)
      .isEquals();
  }

  @Override
  public int hashCode() {
    return Objects.hash(name, versions, current);
  }

  @Override
  public OperationDeviceAction operation() {
    return OperationDeviceAction.GET;
  }

  public String getName() {
    return name;
  }

  public void setName(final String name) {
    this.name = name;
  }

  public Integer getVersions() {
    return versions;
  }

  public void setVersions(final Integer versions) {
    this.versions = versions;
  }

  public Boolean getCurrent() {
    return current;
  }

  public void setCurrent(final Boolean current) {
    this.current = current;
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/audit/entities/GetCredential.kt
package org.cloudfoundry.credhub.audit.entities

import org.apache.commons.lang3.builder.EqualsBuilder
import org.cloudfoundry.credhub.audit.OperationDeviceAction
import org.cloudfoundry.credhub.audit.RequestDetails
import java.util.Objects

class GetCredential : RequestDetails {
    var name: String? = null
    var versions: Int? = null
    var current: Boolean? = null

    constructor(credentialName: String, numberOfVersions: Int?, current: Boolean) : super() {
        name = credentialName
        versions = numberOfVersions
        this.current = current
    }

    constructor() : super() {
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }

        if (other == null || javaClass != other.javaClass) {
            return false
        }

        val that = other as GetCredential?

        return EqualsBuilder()
            .append(name, that!!.name)
            .append(versions, that.versions)
            .append(current, that.current)
            .isEquals
    }

    override fun hashCode(): Int {
        return Objects.hash(name, versions, current)
    }

    override fun operation(): OperationDeviceAction {
        return OperationDeviceAction.GET
    }
}
